cpp/src/arrow/dataset/file_parquet_encryption_test.cc

```diff
@@ -79,14 +83,20 @@ const auto kAllParamValues = ::testing::Values(
     EncryptionTestParam{false, true, true}, EncryptionTestParam{true, true, true});
 
 // Base class to test writing and reading encrypted dataset.
-class DatasetEncryptionTestBase : public testing::TestWithParam<EncryptionTestParam> {
+template <typename T, typename = typename std::enable_if<
+                          std::is_base_of<EncryptionTestParam, T>::value>::type>
```

Comment:
No need to change this, but as a nit it can probably be simplified to
```suggestion
template <typename T, typename = typename std::enable_if_t<
                          std::is_base_of_v<EncryptionTestParam, T>>
```